core-lib/TestSuite/ActorTests.som

```diff
@@ -596,4 +609,223 @@ class ActorTests usingPlatform: platform testFramework: minitest = Value (
   ) : (
     TEST_CONTEXT = ()
   )
+
+  public class PromiseRuin = AsyncTestContext (
+    | private Exception = kernel Exception. aliceValue = 10. bobValue = 20.|
+    )(
+     private class NoExceptionReceivedException signal = Exception (
+        | public msg |
+        msg := 'received value, expected exception'.
+        self signal.
+    )()  
+
+    private class Alice new = (
+      |exception|
+        exception := GenericTestException1 new.
+    )(
+      public method = (^ aliceValue)
+      public methodFailing = (exception signal)
+    )
+
+    private class Bob new: a = (
+      |exception alice promise|
+        alice := a.
+        exception := GenericTestException1 new.
+    )(
+      public send = (^ a <-: method)
+      public sendFailing = (^ a <-: methodFailing)
+      public sendFailingAndHandle = (
+        promise := a <-: methodFailing.
+        ^ promise whenResolved: [:v | NoExceptionReceivedException signal] onError: [:e | bobValue]
+      )
+    )
+
+    (*
+    public testAsyncOnErrorPromiseRuinsToException = (
+      | promisePair wrPromise assertPromise exception |
+      promisePair := actors createPromisePair.
+      exception := GenericTestException1 new.
+      wrPromise := promisePair promise whenResolved: [ :v | NoExceptionReceivedException signal . ] onError: [ :ex | ex . ]. 
+
+      assertPromise := wrPromise whenResolved: [:v | assert: v msg equals: exception msg . ] .
+
+      promisePair ruin: exception.
+      ^ assertPromise
+    ) *)
+
+  (*    
+    public testAsyncPromiseOfPromiseResolvesToException = (
```

Comment:
Ok, so, just to clarify, do you want me to change the things I currently marked as issues and also try to solve the remaining bugs? I could do that. But I don't want to just take it out of your hands, since you probably spend already a lot of sweat on the PR.